### PR TITLE
Add section on `external`

### LINKED ISSUE #5478: Document `external` on the Functions page
The `external` keyword isn't explained anywhere on the site. Right now, references of it point to the spec. 

With [JS interop](https://github.com/dart-lang/site-www/issues/5438) and [extension type](https://github.com/dart-lang/site-www/issues/4177) docs developing, I think it's important to have a high-level summary of `external` to point to when we're talking about it. 

I think the section should go on the [Functions](https://dart.dev/language/functions) page, and be relatively high-level (description in the [spec](https://spec.dart.dev/DartLangSpecDraft.pdf#External%20Functions) is good). Since it's highly platform dependent, we can just summarize it and point to interop docs (maybe use [C](https://dart.dev/interop/c-interop) and [JS](https://dart.dev/interop/js-interop) specifically as an example, until we have a [interop overview](https://github.com/dart-lang/site-www/issues/5439) page to point to instead).

### PR DESCRIPTION
**Pull request by MaryaBelanger**
Fixes #5478

### CODE REVIEW

File: src/language/functions.md
Comment (parlough) [inline src/language/functions.md:521]: This "C" in the raw markdown looks funny for some reason 😆. It's so short. Good thing Dart has 4 letters 👀

File: src/language/functions.md
Comment (parlough) [inline src/language/functions.md:506]: This sentence being imperative makes me slightly concerned developers will misinterpret it as "if you make a function `external` in one file, **you** can implement it in another **Dart** file". Is that worry justified?

If so, would it make sense to instead just mention _how_ they are declared and rely on the various interop docs for actual instructions? Or move up the discussion from the last paragraph about when it appears, since that's the main reason we're documenting it? Not sure 😕 

Maybe I'm overthinking it. I'll think some more but will leave this comment for now to get your thoughts :)

**Review by parlough — COMMENTED** [review]
Another one down _(from the external keyword table links)_ 🥳!

Thanks for working on this Marya! Reads well to me.

I do have one question that is maybe me overthinking things. Let me know what you think :)

File: src/language/functions.md
Comment (MaryaBelanger) [reply src/language/functions.md:521]: > Good thing Dart has 4 letters 👀

Clearly the superior language 😆

File: src/language/functions.md
Comment (MaryaBelanger) [reply src/language/functions.md:506]: No that's a really good point (in fact, it sounds like exactly the kind of question I'd leave in a review of someone else's work haha).

I do want to leave the last paragraph where it is (though your reasoning makes sense, I think since this is purely a reference section the first sentence should be a generic definition), but I think I'll restate the opening more or less as it's stated in the [spec](https://spec.dart.dev/DartLangSpecDraft.pdf#External%20Functions):

> An external function is a function whose body is provided separately from its declaration.

File: src/language/functions.md
Comment (parlough) [inline src/language/functions.md:505]: Starting with this does seem safer, thanks! However, do consider briefly introducing the following code snippet/syntax, otherwise it's not clear why it's there/what it means.

**Review by parlough — APPROVED** [review]
Looks great to me after one more consideration. Thanks for documenting this Marya!

Do consider waiting for Erik's review before landing, unless you need this landed for something else. We can always make changes as the various uses of the keyword start to be documented too :D

File: src/language/functions.md
Comment (eernstg) [inline src/language/functions.md:514]: We don't even have to say 'essentially' --- a final external variable is just a different syntax specifying an external getter, and a non-final external variable is just a different syntax specifying an external getter and an external setter. Also, we can't know whether there is anything out there that matches the description 'external value', it's perfectly possible that such getters/setters are bound to C functions, with no direct connection to anything that resembles a variable. Perhaps the text could be adjusted a bit like this:

```suggestion
An [instance variable][] can be `external` too,
which is equivalent to an external getter and (if the variable
is not `final`) an external setter.
```

It is also possible for a static variable to be external, and a library variable (aka a top-level variable), but not a local variable (and not a function parameter). I don't know if it's important to mention the static/library variants, that would depend on the level of detail/completeness which is customary and desired here.

File: src/language/functions.md
Comment (eernstg) [inline src/language/functions.md:537]:
```suggestion
[instance variable]: /language/classes#instance-variables
```

**Review by eernstg — APPROVED** [review]
LGTM!

**Pull request merged**
